repondeur/zam_repondeur/tasks/fetch.py

```diff
@@ -101,50 +104,75 @@ def fetch_amendements(lecture_pk: Optional[int]) -> bool:
             source.prepare(lecture)
         logger.info("Time to prepare: %.1fs", prepare_timer.elapsed())
 
-        # Collect data about new and updated amendements.
-        with Timer() as collect_timer:
-            changes = source.collect_changes(lecture)
-        logger.info("Time to collect: %.1fs", collect_timer.elapsed())
+        cumulated_result = FetchResult.create()
 
-        # Then apply the actual changes in a fresh transaction, in order to minimize
-        # the duration of database locks (if we hold locks too long, we could have
-        # synchronization issues with the webapp, causing unwanted delays for users
-        # on some interactive operations).
+        nb_batches = 1
+        start_index = 0
 
-        # But during tests we want everything to run in a single transaction that
-        # we can roll back at the end.
-        if not huey.immediate:
-            transaction.commit()
-            DBSession.expire_all()
+        while True:
```

Comment:
Given that we have a `total_timer` maybe we can add a timeout to that loop if it takes too much time?

Reply:
How long is too long?

Reply:
100000/250*120/60/60 => 13h20 if we consider a max loop at 2 minutes (120 seconds) 😰